co-log-core/src/Colog/Core/Action.hs

```diff
@@ -217,6 +218,11 @@ cfilter :: Applicative m => (msg -> Bool) -> LogAction m msg -> LogAction m msg
 cfilter predicate (LogAction action) = LogAction $ \a -> when (predicate a) (action a)
 {-# INLINE cfilter #-}
 
+cfilterM :: Monad m => (a -> m Bool) -> LogAction m a -> LogAction m a
```

Comment:
It would be nice to add some Haddock documentation, saying when this function can be useful. For example, if you want to log messages only on weekends, you can have a monadic predicate like this `isWeekend :: MessageWithTimestamp -> IO Bool` and use it here.